Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/TargetsTests/RoslynTargetsTests.cs

```diff
@@ -831,121 +866,15 @@ private static void AssertExpectedAnalysisSetting(BuildLog actualResult, string
 
         #region Setup
 
-        /// <summary>
-        /// Creates a valid project with the necessary ruleset and assembly files on disc
-        /// to successfully run the "OverrideRoslynCodeAnalysisProperties" target
-        /// </summary>
-        private string CreateProjectFile(AnalysisConfig analysisConfig, string testSpecificProjectXml)
+        private string CreateProjectFile(AnalysisConfig config, string projectSnippet, string afterTargets)
```

Comment:
All invocations uses `AfterTargets` property. So you can remove the argument and use that property inside this method directly.
```suggestion
        private string CreateProjectFile(AnalysisConfig config, string projectSnippet)
```

As `AfterTargets` would be used only inside `CreateProjectFile`, it's a question if you need the property itself and `Initialize()` orchestration to build it. It could also be a local variable.